client/analytics/report/downloads/table.js

```diff
@@ -85,16 +85,31 @@ class CouponsReportTable extends Component {
 				product_id: productId,
 				username,
 			} = download;
-			const { name: productName } = _embedded.product[ 0 ];
 
-			const productLink = getNewPath(
-				persistedQuery,
-				'/analytics/products',
-				{
-					filter: 'single_product',
-					products: productId,
-				}
-			);
+			const { code: errorCode, name: productName } = _embedded.product[ 0 ];
+			let productDisplay, productValue;
+
+			// Handle deleted products.
+			if ( errorCode === 'woocommerce_rest_product_invalid_id' ) {
```

Comment:
Just curious and may not require any change, but wondering if there are cases where we receive an error code for a product but still want to display it.

Reply:
I'm interpreting this as "why even check the code, just check for _any_ error" - is that right?

TBH I don't know the full range of possible error codes here. I think a comprehensive look at deleted objects and their effect on our reports is worth the effort as a follow up.